client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx

```diff
@@ -12,40 +12,75 @@ import { localize } from 'i18n-calypso';
 
-const DisconnectSurvey = ( { confirmHref, isPaidPlan, siteId, siteSlug, translate } ) => (
-	<div className="disconnect-site__survey main">
-		<QuerySitePlans siteId={ siteId } />
-		<SectionHeader
-			label={ translate(
-				'Would you mind sharing why you want to disconnect %(siteName)s from WordPress.com?',
-				{
-					args: { siteName: siteSlug },
-				}
-			) }
-		/>
-		<CompactCard href={ `/settings/disconnect-site/too-difficult/${ siteSlug }` }>
-			{ translate( 'It was too hard to configure Jetpack.' ) }
-		</CompactCard>
-		<CompactCard href={ `/settings/disconnect-site/missing-feature/${ siteSlug }` }>
-			{ translate( 'A feature I need was missing.' ) }
-		</CompactCard>
-		{ isPaidPlan && (
-			<CompactCard href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }>
-				{ translate( 'This plan is too expensive.' ) }
-			</CompactCard>
-		) }
-		<CompactCard href={ confirmHref + '?reason=troubleshooting' }>
-			{ translate( "Troubleshooting — I'll be reconnecting afterwards." ) }
-		</CompactCard>
-		<CompactCard href={ confirmHref + '?reason=other' }>
-			{ translate( 'Another reason…' ) }
-		</CompactCard>
-	</div>
-);
+class DisconnectSurvey extends PureComponent {
+	state = {
+		otherReason: '',
+		hasOtherReasonFocus: false,
+	};
+
+	interceptOtherReasonClick = event => {
+		if ( ! this.state.otherReason || this.state.hasOtherReasonFocus ) {
+			event.preventDefault();
+		}
+	};
+
+	setOtherReason = event => {
+		this.setState( { otherReason: event.target.value } );
+	};
+
+	setOtherReasonFocus = focus => () => {
+		this.setState( { hasOtherReasonFocus: focus } );
+	};
+
+	render() {
+		const { confirmHref, isPaidPlan, siteId, siteSlug, translate } = this.props;
+		return (
+			<div className="disconnect-site__survey main">
+				<QuerySitePlans siteId={ siteId } />
+				<SectionHeader
+					label={ translate(
+						'Would you mind sharing why you want to disconnect %(siteName)s from WordPress.com?',
+						{
+							args: { siteName: siteSlug },
+						}
+					) }
+				/>
+				<CompactCard href={ `/settings/disconnect-site/too-difficult/${ siteSlug }` }>
+					{ translate( 'It was too hard to configure Jetpack.' ) }
+				</CompactCard>
+				<CompactCard href={ `/settings/disconnect-site/missing-feature/${ siteSlug }` }>
+					{ translate( 'A feature I need was missing.' ) }
+				</CompactCard>
+				{ isPaidPlan && (
+					<CompactCard href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }>
+						{ translate( 'This plan is too expensive.' ) }
+					</CompactCard>
+				) }
+				<CompactCard href={ confirmHref + '?reason=troubleshooting' }>
+					{ translate( "Troubleshooting — I'll be reconnecting afterwards." ) }
+				</CompactCard>
+				<CompactCard
+					href={ addQueryArgs( { reason: 'other', text: this.state.otherReason }, confirmHref ) }
+					onClick={ this.interceptOtherReasonClick }
+				>
+					{ translate( 'Another reason:' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Another reason…' )` *ES Score: 19*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).